examples/adaptive_library/TrapezoidalDuct.py

```diff
@@ -0,0 +1,225 @@
+                            )
+                            new_end_x, new_end_y = rt_to_xy(r_end_point, new_angle_end_point)
+                            new_start_point = Coordinate(new_start_x, new_start_y)
+                            new_end_point = Coordinate(new_end_x, new_end_y)
+                            duct_region.edit_point(entity.start, new_start_point)
+                            duct_region.edit_point(entity.end, new_end_point)
+                            mc.set_region(duct_region)
+
+        elif round(duct_region.area / duct_area, 2) == 0.5:  # half duct
+            Symm_angle = 360 / duct_region.duplications  # angle of symmetry
+
+            for i, entity in enumerate(duct_region.entities):
+                if round(entity.length / Trap_W, 2) == 0.5:  # check if  the line is width
+                    # additional check in case width = height
+                    r_start_point, angle_start_point = xy_to_rt(entity.start.x, entity.start.y)
+                    r_end_point, angle_end_point = xy_to_rt(entity.end.x, entity.end.y)
+                    if abs(angle_end_point - angle_start_point) > 0.05:  # 0.05 degree is tolerance
+                        Line_origin = check_line_origin_distance(i, duct_region)
+                        if not Line_origin:
+                            del_trap_angle = (angle_end_point - angle_start_point) * (
+                                1 - Trap_ratio
+                            )
+                            if (
+                                angle_start_point - 0 < 1e-10
+                                or angle_start_point == 0
+                                or round(angle_start_point / Symm_angle, 2) == 1
+                            ):  # on symmetry plane
+                                new_angle_end_point = angle_end_point - del_trap_angle
+                                new_end_x, new_end_y = rt_to_xy(r_end_point, new_angle_end_point)
+                                new_end_point = Coordinate(new_end_x, new_end_y)
+                                duct_region.edit_point(entity.end, new_end_point)
+                            elif (
+                                angle_end_point - 0 < 1e-10
+                                or round(angle_end_point / Symm_angle, 2) == 1
+                            ):  # symmetry plan
+                                new_angle_start_point = angle_start_point + del_trap_angle
+                                new_start_x, new_start_y = rt_to_xy(
+                                    r_start_point, new_angle_start_point
+                                )
+                                new_start_point = Coordinate(new_start_x, new_start_y)
+                                duct_region.edit_point(entity.start, new_start_point)
+
+                            mc.set_region(duct_region)
+
+
+# %%
+# Load in Adaptive Templates Script if required
+# ---------------------------------------------
+# When the script is run externally:
```

Comment:
This isn't clear. Does the script do the following actions or do you have to do them when running the script externally? All UI elements should be in bold.

Reply:
This part is taken from already published examples. The script do the following actions for loading the script externally